## Exercice 1 Mélange éthanol-eau
On mélange 50 g de d'ethanol pur, A, avec 20 g d'eau, W.
Calculer les fractions molaires, massiques et volumique.

Données à 20°C :
$\rho_{eau}=1000 kg/m3$
$\rho_{eth}=789 kg/m3$
$M_{eth}=46 g/mol$
$M_{eau}=18 g/mol$

In [4]:
#Données W eau et A alcool
rho_W=1000  #M_W/V_W kg/m3 ou g/L
rho_A=789   #M_A/V_A kg/m3 ou g/L
M_A=46      #g/mol
M_W=18      #g/mol

m_A=50      #g A
m_W=20      #g W
m_T=m_A+m_W

#Fraction massique
x_mA=m_A/m_T
x_mW=m_W/m_T

#Fraction molaire 
n_A=m_A/M_A
n_W=m_W/M_W
n_T=n_A+n_W
x_nA=n_A/n_T
x_nW=n_W/n_T

#Fraction volumique 
V_A=m_A/rho_A #L
V_W=m_W/rho_W #L
V_T=V_A+V_W   #L on neglige les volumes de mélanges qui seront vus plus tard
x_VA=V_A/V_T
x_VW=V_W/V_T

print('{:<20}{:<8}{:<8}'.format('','Ethanol','Eau'))
print('{:<20}{:.3g}{:<3}{:.3g}'.format('Fraction massique',x_mA,'', x_mW))
print('{:<20}{:.3g}{:<3}{:.3g}'.format('Fraction molaire',x_nA,'', x_nW))
print('{:<20}{:.3g}{:<4}{:.3g} (en négligeant les volumes de mélange)'.format('Fraction volumique',x_VA,'', x_VW))

                    Ethanol Eau     
Fraction massique   0.714   0.286
Fraction molaire    0.495   0.505
Fraction volumique  0.76    0.24 (en négligeant les volumes de mélange)


## Exercice 1
Quels sont les masses à peser d'éthanol et d'eau pour réaliser 20g d'un mélange avec une fraction molaire de 30% en éthanol ?
>Il s'agit ici de déterminer deux inconnues m_A et m_W sachant que :
>- $m_{eth}+m_{eau}=20 g$
>- $n_{eth}/n_T=0.3$
>
> Or $n_{eth}=m_{eth}/M_{eth}$ et $n_W=m_W/M_W$. Le système à résoudre est donc :
>- $m_{eth}+m_{eau}=m_T=20 g$
>- $\frac{m_{eth}/M_{eth}}{m_{eth}/M_{eth}+m_{eau}/M_{eau}}=x_{eth}=0.3$
>
>On est arrivé à exprimer le problème avec deux équations et deux inconnus : c'est raassurant et un premier pas important franchi ! On peut ensuite combiner ces équations pour avoir une équation à une inconnue, choisissons de tout exprimer en fonction de $m_A$ :
>
>$m_{eth}/M_{eth}=x_{eth}(m_{eth}/M_{eth}+(m_T-m_{eth})/M_{eau})$
>
>$m_{eth}/M_{eth}-x_{eth}(m_{eth}/M_{eth}+(m_T-m_{eth})/M_{eau})=0$
>
>$m_{eth}(1/M_{eth}-x_{eth}(1/M_{eth}-1/M_{eau}))-x_{nA}m_T/M_{eau}=0$
>
>$m_{eth}(1/M_{eth}-x_{eth}(1/M_{eth}-1/M_{eau}))=x_{nA}m_T/M_{eau}$
>
>$m_{eth}=\frac{x_{eth}m_T/M_{eau}}{1/M_{eth}-x_{nA}(1/M_{eth}-1/M_{eau})}$
>
>$\frac{m_{eth}}{m_T}=w_{eth}=\frac{1}{1+(1/x_{eth}-1)M_{eau}/M_{eth}}$
>
>Cette dernière écriture permet de vérifier que les unités sont correctes (c'est important pour trouver ses erreurs). 
>
>On peut aussi partir de l'écriture de la fraction massique en supposant que l'on connait le nombre de mole total, $n_T$ qui ensuite à la bonne idée de se simplifier !
>
>$w_{eth}=\frac{m_{eth}}{m_{eth}+m_{eau}}$
>
>$w_{eth}=\frac{x_{eth}~n_T~M_{eth}}{x_{eth}~n_T~M_{eth}+x_{eau}~n_T~M_{eau}}$
>
>$w_{eth}=\frac{x_{eth}~M_{eth}}{x_{eth}~M_{eth}+(1-x_{eth})~M_{eau}}$
>
>C'est un calcul assez compliqué : cela vaut le coup de recalculer la fraction molaire une fois les masses déterminés pour vérifier son calcul.

In [3]:
m_T=20
x_nA=0.3
x_mA=1/(1+(1/x_nA-1)*M_W/M_A)
print('Pour une fraction molaire de', x_nA, "il faut une fraction massique de", round(x_mA,3))
m_A=x_mA*m_T
m_W=m_T-x_mA*m_T
print('il faut peser', round(m_A,3), 'g d\'éthanol et', round(m_W,3), 'g d\'eau')
print('')
n_A=m_A/M_A
n_W=m_W/M_W
print('Vérification nA=',round(n_A,3),' n_W=', round(n_W,3), 'soit une fraction molaire de', round(n_A/(n_A+n_W),2))


Pour une fraction molaire de 0.3 il faut une fraction massique de 0.523
il faut peser 10.455 g d'éthanol et 9.545 g d'eau

Vérification nA= 0.227  n_W= 0.53 soit une fraction molaire de 0.3


# Quantité et composition

Il est important de savoir manipuler et convertir les quantités et les compositions. Dans les procédés, il est aussi souvent nécessaire de faire des bilans sur les quantités et les débits. Cette feuille doit vous permettre de vous exercer et de développer une méthodologie vous permettant d'effecteur ces calculs sans erreurs (80 % des points perdus dans les examens sont liées à des problèmes de calculs de ce type).

## Quantité

On peut representer une quantité de trois façons différentes par : 
- le nombre de **moles**, n, lié aux nombres de molécules et donc d'objets avec le [nombre d'Avogadro](https://fr.wikipedia.org/wiki/Nombre_d%27Avogadro))
- la **masse**, m
- le **volume**, V

En ajoutant un indice, on peut ainsi définir des quantités partielles, c'est à dire des quantités d'un constituant dans un mélange (par exemple, $n_A$ représente le nombre de mole de A). On peut également utilisé l'indice T pour la quantité totale (pour un mélange composé de A et B : $n_T=n_A+n_B$). Les quantités peuvent ainsi s'ajouter.

On peut passer d'une quantité à l'autre en utilisant des rapports de quantité qui caractérise les propriétés des constituants ou des mélanges :
- la **masse molaire** réprésente le rapport la masse par moles, $M_A=\frac{m_A}{n_A}$
- la **masse volumique** représente le rapport la masse par unité de volume $\rho_A=\frac{m_A}{V_A}$
- le **volume molaire** représente le rapport le volume occupé par une mole $v_A=\frac{V_A}{n_A}$

Il faut utiliser avec précaution ces changements de quantité en vérifiant à quelles quantités elles se rapportent, quantité d'un constituant ou du mélange. On peut ainsi définir la masse volumique du constituant A $\rho_A=\frac{m_A}{V_A}$ mais aussi du mélange $\rho_T=\frac{m_T}{V_T}$. Vous pouvez aussi remarquer que ces rapports ne s'aditionnent pas : $\rho_T \ne \rho_A+\rho_B$ mais $V_T \rho_T = V_A \rho_A + V_B \rho_B$ ou $\frac{m_T}{\rho_T} = \frac{m_A}{\rho_A}+\frac{m_B}{\rho_B}$.

## Composition

Les compositions représentent un rapport de quantité dans les mélanges : généralement une quantité d'un constituant sur une quantité totale. On peut utiliser :

Les **concentrations** représentent généralement une quantité d'un constituant sur le volume total, $V_T$. On peut définir :
- une concentration molaire $c_{nA}=n_A/V_T$
- une concentration massique $c_{mA}=m_A/V_T$
- une concentration volumique $c_{VA}=V_A/V_T$ qui est aussi une fraction volumique ou un degré (noté°)

Les **fractions** représentent une quantité d'un constituant sur la même quantité totale. On peut ainsi définir :
- une fraction molaire $x_{A}=n_A/n_T$
- une fraction molaire $w_{A}=m_A/m_T$
- une fraction volumique $\phi_{A}=c_{VA}=V_A/V_T$


**Pour continuer**

Vous pouvez vous exercer à passer d'une fraction massique à une fraction molaire. Si vous avez une fraction massique de 0.523 en ethanol dans l'eau, quel est la fraction molaire ?!